src/Compilers/Core/Desktop/AnalyzerFileReference.cs

```diff
@@ -208,13 +213,12 @@ internal void AddAnalyzers(ImmutableArray<DiagnosticAnalyzer>.Builder builder, s
                 // If there are no analyzers, don't load the assembly at all.
                 if (!analyzerTypeNameMap.ContainsKey(language))
                 {
-                    this.AnalyzerLoadFailed?.Invoke(this, new AnalyzerLoadFailureEventArgs(AnalyzerLoadFailureEventArgs.FailureErrorCode.NoAnalyzers, null, null));
```

Comment:
Another option is to remove the warning but log the names of all the recognized analyzer types by assembly in the error log. That was a request for compliance anyway. It might make the error log too big. Another option is to log the typees on the command line itself under a verbose flag

Sent from my Windows Phone
________________________________
From: Manish Vasani<mailto:notifications@github.com>
Sent: ý4/ý2/ý2015 10:18 AM
To: dotnet/roslyn<mailto:roslyn@noreply.github.com>
Cc: Srivatsn Narayanan<mailto:srivatsn@microsoft.com>
Subject: Re: [roslyn] Don't error when an assembly has no analyzers (#1755)


In src/Compilers/Core/Desktop/AnalyzerFileReference.cs<https://github.com/dotnet/roslyn/pull/1755#discussion_r27677412>:

> @@ -208,13 +213,12 @@ internal void AddAnalyzers(ImmutableArray<DiagnosticAnalyzer>.Builder builder, s
>                  // If there are no analyzers, don't load the assembly at all.
>                  if (!analyzerTypeNameMap.ContainsKey(language))
>                  {
> -                    this.AnalyzerLoadFailed?.Invoke(this, new AnalyzerLoadFailureEventArgs(AnalyzerLoadFailureEventArgs.FailureErrorCode.NoAnalyzers, null, null));


I actually think it at least deserves to be a Warning. Skipping running analyzers when provided with a true analyzer assembly with analyzers, is a bad false positive about the project being clean against analyzers. Keeping it as info is likely going to cause it to never show up on the console output for command line builds, and not show in the output window by default.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/1755/files#r27677412>.